src/components/withParentReportAction.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+import ONYXKEYS from '../ONYXKEYS';
+import reportActionPropTypes from '../pages/home/report/reportActionPropTypes';
+import reportPropTypes from '../pages/reportPropTypes';
+
+const withParentReportActionPropTypes = {
+    parentReportAction: PropTypes.shape(reportActionPropTypes),
+};
+
+const withParentReportActionDefaultProps = {
+    parentReportAction: {},
+};
+
+export default function (WrappedComponent) {
+    const propTypes = {
+        forwardedRef: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({current: PropTypes.instanceOf(React.Component)})]),
+
+        /** All report actions for the parent report */
+        parentReportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+        /** The report currently being looked at */
+        report: reportPropTypes,
+    };
+    const defaultProps = {
+        forwardedRef: undefined,
+        report: {},
+        parentReportActions: {},
+    };
+    function WithParentReportAction(props) {
+        const parentReportActionID = props.report && props.report.parentReportActionID ? props.report.parentReportActionID : {};
```

Comment:
Ah why are we setting an empty object for `parentReportActionID`? Additionally, shouldn't we add `parentReportActionID` to `reportPropTypes`?